### DISCUSSION TITLE
Is the BTT smart sensor compatible with this firmware?

### DISCUSSION THREAD

**Discussion by SpikeMeester** [Jyers/Marlin#581] [Q&A] [answered]
With default 4.2.2 firmware the BTT sensor triggers filament out just on purge line but if I hit the reload button it will carry on with the print and I get no more filament out errors????....Checked on in menu with 7mm runout and saved with SD card in?????

**Comment by Cal-LF** [comment, marked as answer]
I found that my BTT sensor will work but have to self compile from one of the configuration templates. Have to un-comment line1347 #define FILAMENT_MOTION_SENSOR in the config.h file. It can then be selected in the advanced menu screen. I use 10mm runout because the recommended 7mm was getting false trips.

**Reply by SpikeMeester** [reply]
For someone who has the ability to do just that it sounds simple enough, but for us that don't have that skill should that not be an option in the firmware so we can use the BTT sensor?

**Comment by holsteiner-kiel** [comment]
That would be very nice to have option for BTT smart sensor directly in your firmware. BTW: Very goog work. Backed your project a bit :-)

**Reply by bkarr123** [reply]
Same! This option would be great to have, is it something you have considered for future precompiled versions?

**Reply by Cal-LF** [reply]
This question has been answered in previous discussions. Can only be activated by compiling your own FW.

**Comment by smooreace** [comment]
Its just Marlin FW with screen tweaks. Compile is however you need it to be

**Reply by gothcha** [reply]
The options like motion sensor are controlled by #defines in the firmware, which controls the code that is included during compilation, it is not possible to enable in software at run time unfortunately.
